### PR TITLE
Implement Connector Builder testing panel (first iteration)

### PR DESCRIPTION
**Pull request by lmossman**
## What
Relates to https://github.com/airbytehq/airbyte/issues/17917 (this PR sets up most of the scaffolding for the testing panel -- the next PR will focus on styling the testing panel results display to bring it closer to the [design](https://www.figma.com/proto/F1ljvrJerLoVbNvtejBWid/LOW-CODE-CONNECTOR-BUILDER?node-id=1926%3A72640&scaling=min-zoom&page-id=1926%3A72574&starting-point-node-id=1926%3A72575))

Here is a screenshot of what this first iteration looks like after `Test` has been clicked:
Screen Shot 2022-10-25 at 3 48 48 PM

## How
This PR adds the implementation of the testing side-panel to the connector builder UI. 

The main changes in this PR are as follows:
1. The pieces of state that need to be shared around multiple places in the Connector Builder have been encapsulated in the `ConnectorBuilderStateService`, so that the state does not need to be passed around the components through props
2. Sets up the components in the testing side panel to implement its basic functions and layout
3. Wires up the ConnectorBuilderApiService to the components along with the ConnectorBuilderStateService, so that clicking the Test button actually submits the manifest, config, and selected stream to the Api service, and then displays the mocked result that that returns.

I've also left some comments in the PR with some more explanations of other changes.

## Recommended reading order
1. `ConnectorBuilderStateService.tsx` - contains the logic to store and manipulate the shared state for this UI, as well as the logic for converting the manifest between yaml and json
2. `StreamTestingPanel.tsx`- the top-level component implementing the testing side-panel. See the child components here for the full implementation of the side panel
3. `CodeEditor.tsx` - changed the CodeEditor component to be more generic, as it is now used by the `YamlEditor` and `ConfigMenu` components

## 🚨 User Impact 🚨
Only impact is to the connector builder UI, which is only accessible in OSS by going to a specific route that is not publicized, so there shouldn't be any user impact from this PR

### CODE REVIEW

File: airbyte-webapp/src/components/StreamTestingPanel/ResultDisplay.tsx
Comment (lmossman) [inline airbyte-webapp/src/components/StreamTestingPanel/ResultDisplay.tsx:13]: This component is not at all finalized -- this was just a quick and dirty implementation to display all of the contents of the `StreamRead` object in order to show that the data was successfully retrieved and give a hint at what it will look like.

This component will be the focus of the next PR, in which I will work on bringing it closer to the [design](https://www.figma.com/proto/F1ljvrJerLoVbNvtejBWid/LOW-CODE-CONNECTOR-BUILDER?node-id=1926%3A72640&scaling=min-zoom&page-id=1926%3A72574&starting-point-node-id=1926%3A72575)

File: airbyte-webapp/src/components/ui/CodeEditor/CodeEditor.tsx
Comment (lmossman) [inline airbyte-webapp/src/components/ui/CodeEditor/CodeEditor.tsx:16]: This component was modified to be more generic, since I am now using it in two places (YamlEditor and ConfigMenu)

File: airbyte-webapp/src/pages/ConnectionPage/pages/ConnectionItemPage/ConnectionStatusTab.tsx
Comment (lmossman) [inline airbyte-webapp/src/pages/ConnectionPage/pages/ConnectionItemPage/ConnectionStatusTab.tsx:180]: This change and the SCSS changes for this component are to fix the styling of the Sync button on the connection page.

Before:
Screen Shot 2022-10-25 at 2 47 18 PM

After:
Screen Shot 2022-10-25 at 2 47 31 PM

File: airbyte-webapp/src/components/StreamTestingPanel/StreamTestingPanel.tsx
Comment (lmossman) [inline airbyte-webapp/src/components/StreamTestingPanel/StreamTestingPanel.tsx:12]: FYI: "manifest" is the term we are now using for the low-code YAML connector definition file, as discussed in this slack thread: https://airbytehq-team.slack.com/archives/C03TP6W8081/p1666214768391589

File: airbyte-webapp/src/services/connectorBuilder/ConnectorBuilderStateService.tsx
Comment (lmossman) [inline airbyte-webapp/src/services/connectorBuilder/ConnectorBuilderStateService.tsx:26]: I created this context to hold all of the connector builder state that needs to be set and accessed in multiple places around the application. Currently, that is just the connector manifest value, the testing config JSON blob, and the selected stream. 

I expect that we may need to split this context out into multiple different ones at some point as we add more complexity to the app, but I decided to keep it in one context for now since it is still fairly simple.

File: airbyte-webapp/src/components/StreamTestingPanel/StreamSelector.tsx
Comment (timroes) [inline airbyte-webapp/src/components/StreamTestingPanel/StreamSelector.tsx:12]:
```suggestion
export const StreamSelector: React.FC = () => {
```

File: airbyte-webapp/src/services/connectorBuilder/ConnectorBuilderStateService.tsx
Comment (timroes) [inline airbyte-webapp/src/services/connectorBuilder/ConnectorBuilderStateService.tsx:78]: :heart: for reminding the PropsWithChildren :)

File: airbyte-webapp/src/components/StreamTestingPanel/TestControls.tsx
Comment (timroes) [inline airbyte-webapp/src/components/StreamTestingPanel/TestControls.tsx:29]: This feels weird that this is needed, and sounds like a bug in our Button component. Maybe you could share that with the platform workflow team that they could have a look if we can fix the Button component to not require this wrapped `div` anymore. (Shouldn't block merging this though!)

File: airbyte-webapp/src/services/connectorBuilder/ConnectorBuilderStateService.tsx
Comment (timroes) [inline airbyte-webapp/src/services/connectorBuilder/ConnectorBuilderStateService.tsx:40]: Could make this a `console.error` instead.

File: airbyte-webapp/src/services/connectorBuilder/ConnectorBuilderStateService.tsx
Comment (timroes) [inline airbyte-webapp/src/services/connectorBuilder/ConnectorBuilderStateService.tsx:71]: Same here, could be `console.error` as lot.

**Review by timroes — APPROVED** [review]
Overall looks good to me. Tested locally. I left some comments on code and styling suggestions, but given that it seems to work as expected I'll already approve.

File: airbyte-webapp/src/components/StreamTestingPanel/ConfigMenu.module.scss
Comment (lmossman) [reply airbyte-webapp/src/components/StreamTestingPanel/ConfigMenu.module.scss:9]: I ended up just using the Modal component's `size` prop in order to set the width, and that seems to react to narrow screen widths as desired (it shrinks so that it is fully contained by the screen even for a narrow viewport). Updated the height to `60vh` since that felt right

File: airbyte-webapp/src/components/StreamTestingPanel/StreamSelector.module.scss
Comment (lmossman) [reply airbyte-webapp/src/components/StreamTestingPanel/StreamSelector.module.scss:52]: They are not necessary - I'm not sure why I added this 🤷 good catch - I've removed them

File: airbyte-webapp/src/components/StreamTestingPanel/StreamTestingPanel.tsx
Comment (lmossman) [reply airbyte-webapp/src/components/StreamTestingPanel/StreamTestingPanel.tsx:25]: Agreed - I actually did something like that in the follow up PR. But I will fix that here too

File: airbyte-webapp/src/components/StreamTestingPanel/TestControls.module.scss
Comment (lmossman) [reply airbyte-webapp/src/components/StreamTestingPanel/TestControls.module.scss]: Removed the custom styling here to unblock this, but I'll let @Upmitt comment on if we really want a different button color here

File: airbyte-webapp/src/components/StreamTestingPanel/TestControls.tsx
Comment (lmossman) [reply airbyte-webapp/src/components/StreamTestingPanel/TestControls.tsx:30]: Created this issue: https://github.com/airbytehq/airbyte/issues/18581

File: airbyte-webapp/src/components/StreamTestingPanel/StreamSelector.module.scss
Comment (Upmitt) [reply airbyte-webapp/src/components/StreamTestingPanel/StreamSelector.module.scss]: @timroes I updated the main component on Figma, so we can test it in the prototypes. I think GREY 100 is very dark for a hover state, I would keep the grey 50 but let you test

File: airbyte-webapp/src/components/StreamTestingPanel/TestControls.module.scss
Comment (Upmitt) [reply airbyte-webapp/src/components/StreamTestingPanel/TestControls.module.scss]: @timroes @lmossman the dark button is part of the different button styles in our components library - The goal was to have an important CTA, but keep only the "main" CTA (save) as the primary one. I tested in the prototype with the Primary button (blue) and it works well, so let's use the blue!

Comment (Upmitt) [reply airbyte-webapp/src/components/StreamTestingPanel/TestControls.module.scss]: https://www.figma.com/file/gMNRYVJav25pdsoxAyW7Qp/00_01_WEB-APP-LIBRARIES?node-id=58%3A126

File: airbyte-webapp/src/components/StreamTestingPanel/StreamSelector.module.scss
Comment (lmossman) [reply airbyte-webapp/src/components/StreamTestingPanel/StreamSelector.module.scss]: I also think grey-100 feels a little too dark for hover state. Would it be possible for us to add a new grey color in between 50 and 100?

Comment (timroes) [reply airbyte-webapp/src/components/StreamTestingPanel/StreamSelector.module.scss]: Had a Convo with Nico about that offline. Let's stick to 50 now and address that if so separate from this PR.

**Pull request merged**
